temporalcli/commands.workflow.go

```diff
@@ -22,8 +32,52 @@ func (*TemporalWorkflowResetBatchCommand) run(*CommandContext, []string) error {
 	return fmt.Errorf("TODO")
 }
 
-func (*TemporalWorkflowSignalCommand) run(*CommandContext, []string) error {
-	return fmt.Errorf("TODO")
+func (c *TemporalWorkflowSignalCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	// Get input payloads
+	input, err := c.buildRawInputPayloads()
+	if err != nil {
+		return err
+	}
+
+	exec, batchReq, err := c.workflowExecOrBatch(cctx, c.Parent.Namespace, cl)
+	if err != nil {
+		return err
+	}
+
+	// Run single or batch
+	if exec != nil {
+		// We have to use the raw signal service call here because the Go SDK's
+		// signal call doesn't accept multiple arguments.
+		_, err = cl.WorkflowService().SignalWorkflowExecution(cctx, &workflowservice.SignalWorkflowExecutionRequest{
+			Namespace:         c.Parent.Namespace,
+			WorkflowExecution: &common.WorkflowExecution{WorkflowId: c.WorkflowId, RunId: c.RunId},
+			SignalName:        c.Name,
+			Input:             input,
+			Identity:          clientIdentity(),
+		})
+		if err != nil {
+			return fmt.Errorf("failed signalling workflow: %w", err)
+		}
+		cctx.Printer.Println("Signal workflow succeeded")
+	} else if batchReq != nil {
+		batchReq.Operation = &workflowservice.StartBatchOperationRequest_SignalOperation{
+			SignalOperation: &batch.BatchOperationSignal{
+				Signal:   c.Name,
+				Input:    input,
+				Identity: clientIdentity(),
+			},
+		}
+		if err := startBatchJob(cctx, cl, batchReq); err != nil {
+			return err
+		}
+	}
```

Comment:
I believe that the contract of `workflowExecOrBatch` is such that one or other of the above conditions must be true, so I'd like the code not to appear to allow neither to be executed.

Reply:
Yes, this is leftover from previous code where it could return all nils when you declined the prompt, but I decided to change that to be an error and did not change this.